src/control/cmd/daos_server/storage_nvme.go

```diff
@@ -131,6 +156,51 @@ func (cmd *prepareNVMeCmd) WithPCIAllowList(al string) *prepareNVMeCmd {
 	return cmd
 }
 
+func (cmd *prepareNVMeCmd) WithIgnoreConfig(b bool) *prepareNVMeCmd {
+	cmd.IgnoreConfig = b
+	return cmd
+}
+
+func processNVMePrepReq(log logging.Logger, cfg *config.Server, iommuChecker hardware.IOMMUDetector, req *storage.BdevPrepareRequest) error {
+	if err := sanitizePCIAddrLists(req); err != nil {
+		return errors.Wrap(err, "sanitizing cli input pci address lists")
+	}
+
+	if err := updateNVMePrepReqFromConfig(log, cfg, req); err != nil {
+		return errors.Wrap(err, "updating request parameters with config file settings")
+	}
+
+	iommuEnabled, err := iommuChecker.IsIOMMUEnabled()
+	if err != nil {
+		return errors.Wrap(err, "verifying iommu capability on host")
+	}
+
+	targetUser, err := getTargetUser(req.TargetUser)
+	if err != nil {
+		return err
+	}
+	// Update target user parameter in request.
+	req.TargetUser = targetUser
+
+	if err := validateVFIOSetting(targetUser, req.DisableVFIO, iommuEnabled); err != nil {
+		return err
+	}
+
+	switch {
```

Comment:
Do not understand why all this case should be exclusive ?

Reply:
For each of those cases (in order of priority), VMD will be disabled (with an informational message logged) and the default case skipped (which results in req.EnableVMD retaining its zero value, false).

Reply:
OK